src/Core/CSharpVB/NuGetPackageInfoGenerator.cs

```diff
@@ -49,7 +49,7 @@ public static class NuGetPackageInfoGenerator
         private static IEnumerable<string> GetPluginPropertyPrefixes(IEnumerable<SonarQubeRule> rules) =>
             rules.Select(r => r.TryGetRoslynPluginPropertyPrefix())
                 .Distinct()
-                .Where(r => !string.IsNullOrEmpty(r))
+                .Where(r => !string.IsNullOrEmpty(r) && !RoslynPluginRuleKeyExtensions.IsExcludedRuleRepository(r))
```

Comment:
There is a duplication here.
Both NugetPackageInfoGenerator and RuleSetGenerator go through the same process: transforming active rules into repository keys and then filtering.
I think it should be done in one place, in a higher caller.
It would simplify the process and the tests, as we won't have to check everything twice.
WDYT?

Reply:
As discussed offline, I agree there's a duplication in the filtering logic but I don't think it's worth trying to change now.